samples/net/gsm_modem/src/main.c

```diff
@@ -114,8 +114,6 @@ int main(void)
 {
 	const struct device *uart_dev = DEVICE_DT_GET(UART_NODE);
```

Comment:
nit: this can be also constified.

Reply:
I've only tackled statically defined variables in this PR (just to scope the changes a little bit)